### DISCUSSION TITLE
Built-in function to iterate over a range of sequential numbers

### DISCUSSION THREAD

**Discussion by mitchgrout** [slint-ui/slint#4708] [Q&A] [answered]
Most of the examples of creating a variable number of elements I have seen in the documentation are of this form:

```js
for item[index] in collection : SomeElement { ... }
```

I would like to do something rather simple; drawing gridlines on a rectangle. The first way I think to do it would be to iterate over a range of numbers `0 ... <num-gridlines>`, and using that determine my x/y ordinates for the arc. As far as I can see, there is no method like `@range(min, max, step)` that I could use to express this using a `for`.

If not already existing, would this be a valid case for its addition? For example, something like...
```js
for _[index] in @range(10) : Path { ... } // Similar to Python's range() method
```

**Comment by mitchgrout** [comment, marked as answer]
Ah, found it noted in the `Syntax > Repetition` reference:

> an integer, in which case the element will be repeated that amount of time

This is definitely not expected, `for index in 10` immediately looks like a syntax error to me.

**Reply by tronical** [reply]
Yes, an integer can act as a model. But we could indeed also have a `@range()` macro, like `@range(20, 40)` that goes from `20..<40` for example.

**Reply by bjorn** [reply]
Just wondering, but why would this not be literally written as `20..<40` in the Slint language? It looks cleaner and clearer to me than `@range(20, 40)`. Or is it much more complicated to implement it like that?

**Reply by mitchgrout** [reply]
My pick for `@range` is primarily due to it being a common method name for this pattern in other languages, and it would allow for the flexibility of a third 'step' option, like `@range(x0, x1, distance-between-points)`

**Reply by bjorn** [reply]
Ah, I hadn't seen it before. A least Rust, Ruby and Kotlin all feature a range expression, but yeah it does not allow specifying a step size.
